### PR TITLE
fix: make datastore OS agnostic (path things)

### PR DESCRIPTION
**Pull request by daviddias**

### CODE REVIEW

File: src/key.js [hunk 1]
```diff
@@ -1,11 +1,11 @@
 /* @flow */
 'use strict'
 
-const path = require('path')
 const uuid = require('uuid/v4')
 
-const pathSepS = path.sep
-const pathSep = new Buffer(pathSepS, 'utf8')[0]
+const pathSepS = '/'
+const pathSepB = new Buffer(pathSepS, 'utf8')
```
Comment (daviddias) [inline src/key.js]: needs to change to `Buffer.from`

File: test/key.spec.js [hunk 2]
```diff
@@ -3,16 +3,14 @@
 'use strict'
 
 const expect = require('chai').expect
-const path = require('path')
 
 const Key = require('../src').Key
 
-const pathSep = path.sep
-const n = (p) => path.normalize(p)
+const pathSep = '/'
```
Comment (daviddias) [inline test/key.spec.js:9]: If this is defined in multiple places, it would be better to have a `src/constants.js` file

**Review by daviddias — COMMENTED** [review]
@richardschneider A couple of nitpicks, otherwise LGTM.

File: test/key.spec.js (hunk 2 above)
Comment (richardschneider) [reply test/key.spec.js:9]: I'm not happy with the names, just dealing with the current code.  You will note that `pathSep` is a Buffer in one and a string in another!

File: src/key.js (hunk 1 above)
Comment (richardschneider) [reply src/key.js]: see the following comment

File: test/key.spec.js (hunk 2 above)
Comment (daviddias) [reply test/key.spec.js:9]: You are welcome to change that :)

File: src/key.js (hunk 1 above)
Comment (daviddias) [reply src/key.js]: What I'm referencing here is the linting issue https://travis-ci.org/ipfs/interface-datastore/jobs/296689734#L2153

File: test/key.spec.js (hunk 2 above)
Comment (daviddias) [reply test/key.spec.js:9]: @richardschneider you should be able to fetch this branch and push commits to it.

**Comment by richardschneider** [comment]
@diasdavid  Is the codecov error a false positive?

**Comment by daviddias** [comment]
@richardschneider it is just because it is the first time it is running on this repo. It's fine :)

**Review by richardschneider — APPROVED** [review]
LGTM

**Comment by daviddias** [comment]
@richardschneider released as 0.4.0. You can now update in the remaining PR's \o/

**Comment by richardschneider** [comment]
Cool, thanks for all help

**Comment by daviddias** [comment]
No, thank you for making it happen!! ❤️

**Comment by pvh** [comment]
Looks great guys, I'll take it for a spin today and let you know if I see any other trouble.

**Pull request merged**